ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala

```diff
@@ -207,6 +217,91 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
         }
       }
     }
+
+    "setting dedup context" should {
+      val deduplicateUntil = protobuf.Timestamp.newBuilder().setSeconds(30).build()
+      val submissionTime = protobuf.Timestamp.newBuilder().setSeconds(60)
+
+      def buildContextAndTransaction(
```

Comment:
I think having this functions defined here reduces readability; can you move them at the class or companion level?

Reply:
While I do agree it makes things harder to read I'm not sure if moving the at the class level would improve anything.
The added benefit of having them here is that we have them scoped to the tests where they're used. So it gives a form of context, while having everything at the class/object level just makes me jump around there to read any definition used in the test, and with time it also starts getting quite cluttered, having tens of variables and functions which make no sense.

Reply:
Scala lets you do things in many different ways and there are many opinions about everything. Mine is that I usually try to understand top-down, i.e. from high-level to details, so anything in the way distracts me and I prefer to have good names than the definition at hand. I agree it's opinable, so this too is just a nit.

Reply:
Agree it's nicer to have the tests clean to read. Moved the functions to the bottom and kept only the variables which are used by this tests.